planning/behavior_velocity_planner/autoware_behavior_velocity_crosswalk_module/src/scene_crosswalk.cpp

```diff
@@ -1565,4 +1560,54 @@ bool CrosswalkModule::checkRestartSuppression(
   return planner_param_.min_dist_to_stop_for_restart_suppression < dist_to_stop &&
          dist_to_stop < planner_param_.max_dist_to_stop_for_restart_suppression;
 }
+
+SafetyFactorArray CrosswalkModule::createSafetyFactorArray(
+  const std::optional<StopPoseWithObjectUuids> & stop_factor) const
+{
+  SafetyFactorArray safety_factors;
+  safety_factors.header.stamp = clock_->now();
+  safety_factors.header.frame_id = "map";
+
+  for (const auto & object_id : stop_factor.value().checked_object_ids) {
+    autoware_internal_planning_msgs::msg::SafetyFactor safety_factor;
+
+    safety_factor.type = autoware_internal_planning_msgs::msg::SafetyFactor::OBJECT;
+    safety_factor.object_id = object_id;
+
+    // TODO(odashima): add a predicted path used for the decision
+
+    if (object_info_manager_.objects.count(object_id) > 0) {
+      const auto & object = object_info_manager_.objects.at(object_id);
+      safety_factor.ttc_begin = object.collision_point->time_to_collision;
+      // TODO(odashima): add a correct value
+      safety_factor.ttc_end = object.collision_point->time_to_collision;
+
+      const auto & position = object.position;
+      safety_factor.points = {position};
+    } else {
+      const auto objects_ptr = planner_data_->predicted_objects;
+      for (const auto & object : objects_ptr->objects) {
+        if (object.object_id == object_id) {
+          // TODO(odashima): add a correct value
+          safety_factor.ttc_begin = 0.0;
+          safety_factor.ttc_end = 0.0;
+
+          const auto & position = object.kinematics.initial_pose_with_covariance.pose.position;
+          safety_factor.points = {position};
+          break;
+        }
+      }
```

Comment:
If object is not found in object_info_manager, the object will not be used for the decision. So you can remove this code.